Client/trunk/ParaEngineClient/NPL/NPLHelper.cpp

```diff
@@ -358,80 +358,221 @@ bool NPLHelper::SerializeToSCode(const char* sStorageVar, const luabind::object&
 		STableStack thisRecursionTable = { &input, pRecursionTable};
 		
 		sCode.append("{");
-		int nNumberIndex = 1;
-		for (luabind::iterator itCur(input), itEnd; itCur != itEnd; ++itCur)
+
+		if (sort)
 		{
-			// we only serialize item with a string key
-			const object& key = itCur.key();
-			if (type(key) == LUA_TSTRING)
+			struct sortItem
 			{
-				const char* sKey = object_cast<const char*>(key);
-				const object& value = *itCur;
-				int nOldSize = (int)(sCode.size());
-				// if sKey contains only alphabetic letters, we will use sKey=data,otherwise, we go the safer one ["sKey"]=data.
-				// the first is more efficient in disk space. 
-				int nSKeyCount = (int)strlen(sKey);
-				bool bIsIdentifier = NPLParser::IsIdentifier(sKey, nSKeyCount);
-				if (bIsIdentifier && nSKeyCount > 0)
+				sortItem(const object& key_, const object& value_) : key(key_), value(value_) {}
+				sortItem() {};
+				object key;
```

Comment:
try to optimize performance here. 
Because each new `object` will create a new object on the registry. 

Maybe just using vector<string>?All other keys preserve their native order, and then serialize all string keys in order.